### PR TITLE
Update CHANGELOG for v1.4

### PR DESCRIPTION
**Pull request by lorensr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,57 @@ All notable changes to this project will be documented in this file.
...
+
+- Delete search attributes with empty array values in describe() response ([#878](https://github.com/temporalio/sdk-typescript/pull/878))
+
+  :warning: This fixes a bug where empty/deleted Custom Search Attributes were returned as `[]` from [`workflowHandle.describe()`](https://typescript.temporal.io/api/interfaces/client.workflowhandle/#describe). Such attribute properties will no longer be present in the [`WorkflowExecutionDescription.searchAttributes`](https://typescript.temporal.io/api/interfaces/client.WorkflowExecutionDescription#searchattributes) object. Note that this behavior is consistent with what you'll see if using a pre-1.4 version of the SDK with Server version 1.18.
+
+- Add support for custom failure converters ([#887](https://github.com/temporalio/sdk-typescript/pull/887))
+
+  Adds [`DataConverter.failureConverterPath`](https://typescript.temporal.io/api/interfaces/worker.dataconverter/#failureconverterpath) and [`FailureConverter`](https://typescript.temporal.io/api/interfaces/common.FailureConverter), which converts from proto Failure instances to JS Errors and back.
```
Comment (bergundy) [inline CHANGELOG.md:23]: Let's mark the API as experimental for now and make note of that here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,57 @@ All notable changes to this project will be documented in this file.
 
 Breaking changes marked with a :boom:
 
+## [1.4.0] - 2022-09-28
+
+### Features
+
+- :boom: Make client gRPC retry more configurable ([#879](https://github.com/temporalio/sdk-typescript/pull/879))
+
+  BREAKING CHANGE: [`GrpcRetryOptions.retryableDecider`](https://typescript.temporal.io/api/interfaces/client.grpcretryoptions/#retryabledecider) now gets the `attempt` number as the first argument. This is an advanced/rare option, and the change should be caught at compile time.
```
Comment (bergundy) [inline CHANGELOG.md:13]: I'd note that this API is experimental

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,57 @@ All notable changes to this project will be documented in this file.
...
+
+### Miscellaneous Tasks
+
+- Remove `internal-*` packages ([#881](https://github.com/temporalio/sdk-typescript/pull/881))
+
+  :warning: Any imports from `@temporalio/internal-*` need to be updated. As noted in their named and READMEs, they're not meant to be used to directly, so we don't imagine this is a common case. However, if you do find instances, they should be changed to importing from:
+
+  ```
+  @temporalio/common/lib/internal-non-workflow
+  @temporalio/common/lib/internal-workflow
```
Comment (bergundy) [inline CHANGELOG.md:44]: This isn't a thing, it's just `@temporalio/common`

File: packages/core-bridge/index.d.ts [hunk 4]
```diff
@@ -110,13 +110,12 @@ export interface PrometheusMetricsExporter {
 
 /**
  * Metrics exporters supported by Core
+ *
+ * `temporality` is the type of aggregation temporality for metric export. Applies to both Prometheus and OpenTelemetry exporters.
+ *
+ * See the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/ce50e4634efcba8da445cc23523243cb893905cb/specification/metrics/datamodel.md#temporality) for more information.
```
Comment (bergundy) [inline packages/core-bridge/index.d.ts:116]: Why?

File: packages/workflow/src/interfaces.ts [hunk 5]
```diff
@@ -140,7 +140,7 @@ export interface UnsafeWorkflowInfo {
    * Current system time in milliseconds
    *
    * The safe version of time is `new Date()` and `Date.now()`, which are set on the first invocation of a Workflow
-   * Task.
+   * Task and stay constant.
```
Comment (bergundy) [inline packages/workflow/src/interfaces.ts:143]: What do you mean stay constant? This is inaccurate.

File: packages/core-bridge/index.d.ts (hunk 4 above)
Comment (lorensr) [reply packages/core-bridge/index.d.ts:116]: previous comment location doesn't show up in api ref:

image

Comment (bergundy) [reply packages/core-bridge/index.d.ts:116]: :(

File: packages/workflow/src/interfaces.ts (hunk 5 above)
Comment (lorensr) [reply packages/workflow/src/interfaces.ts:143]: per-Task constant/consistent, vs system/real.

is this clearer?

`stay constant for the duration of the Task and during replay. `

Comment (bergundy) [reply packages/workflow/src/interfaces.ts:143]: Yes

File: packages/common/src/converter/data-converter.ts [hunk 6]
```diff
@@ -38,6 +38,8 @@ export interface DataConverter {
    * Path of a file that has a `failureConverter` named export.
    * `failureConverter` should be an object that implements {@link FailureConverter}.
    * If no path is provided, {@link defaultFailureConverter} is used.
+   *
+   * @experimental
```
Comment (bergundy) [inline packages/common/src/converter/data-converter.ts:42]: Can you please mark the payload converter interface and default implementation as experimental too?

**Review by bergundy — APPROVED** [review]

**Pull request merged**